src/lib/layouts/DocsArticle.svelte

```diff
@@ -13,14 +13,18 @@
 </script>
 
 <script lang="ts">
-    import { scrollToTop } from '$lib/actions/scrollToTop';
-
+    import { setContext } from 'svelte';
     import { Feedback } from '$lib/components';
+    import { scrollToTop } from '$lib/actions/scrollToTop';
 
     export let title: string;
     export let toc: Array<TocItem>;
     export let back: string | undefined = undefined;
     export let date: string | undefined = undefined;
+
+    let reducedArticleSize = false;
+
+    setContext('articleHasNumericBadge', () => (reducedArticleSize = true));
```

Comment:
you can do something like this to make it easier:

```suggestion
    const reducedArticleSize = setContext('articleHasNumericBadge', writable(false));
```

and then on the section components you can do:

```js
if (hasContext('articleHasNumericBadge')) getContext('articleHasNumericBadge').set(true);
```